wgpu-hal/src/lib.rs

```diff
@@ -2802,3 +2803,21 @@ pub struct TlasInstance {
     pub mask: u8,
     pub blas_address: u64,
 }
+
+#[cfg(dx12)]
+pub enum D3D12ExposeAdapterResult {
+    CreateDeviceError(dx12::CreateDeviceError),
+    ResourceBindingTier2Requirement,
+    ShaderModel6Requirement,
+    Success(dx12::FeatureLevel, dx12::ShaderModel),
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct Telemetry {
```

Comment:
We still should add some commentary here about what "Telementry" means here, as it's a bit of a dirty word :)